src/MongoDB.Driver.Core/Core/Authentication/DefaultAuthenticator.cs

```diff
@@ -114,7 +114,10 @@ public async Task AuthenticateAsync(IConnection connection, ConnectionDescriptio
         /// <inheritdoc/>
         public BsonDocument CustomizeInitialIsMasterCommand(BsonDocument isMasterCommand)
         {
-            return isMasterCommand.Merge(CreateSaslSupportedMechsRequest(_credential.Source, _credential.Username)); 
+            var saslSupportedMechs = CreateSaslSupportedMechsRequest(_credential.Source, _credential.Username);
+            isMasterCommand = isMasterCommand.Merge(saslSupportedMechs);
+            var authenticator = new ScramSha256Authenticator(_credential, _randomStringGenerator);
```

Comment:
Default to SCRAM-SHA-256 when the user doesn't provide us a mechanism
mongodb/specifications@e567248#diff-eea9ff495f09b30078faf7f4835e2f14R261